src/cryptography/hazmat/primitives/asymmetric/ec.py

```diff
@@ -245,6 +254,14 @@ def __init__(self, algorithm):
     algorithm = utils.read_only_property("_algorithm")
 
 
+@utils.register_interface(EllipticCurveKeyExchangeAlgorithm)
+class ECDH(object):
```

Comment:
I'm mostly worried about checking that the public key is actually on the same curve as the private key. Malicious public keys are a problem for regular DH, but I'm not sure of their effect on ECDH.

It's probably OK to extract this info from the private key though.